api/v1/installation_types.go

```diff
@@ -202,6 +202,12 @@ type InstallationSpec struct {
 	// Azure is used to configure azure provider specific options.
 	// +optional
 	Azure *Azure `json:"azure,omitempty"`
+
+	// Proxy is used to configure the HTTP(S) proxy settings that will be applied to Tigera containers that connect
+	// to destinations outside the cluster. It is expected that NO_PROXY is configured such that destinations within
+	// the cluster (including the API server) are exempt from proxying.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
```

Comment:
Two questions around validation:

- Since it is our expectation that the user sets no_proxy for in-cluster destinations, do we consider it a misconfiguration if they don't? In other words, should we have validation for it?
- It looks like the env variables on the pod are set based on the proxy settings in the Installation spec and that the egress rules are based on the actual env variables that are on the pod. That's very clever, that means that whether the variables are set by the installation CR or by a mutating webhook, you end up allowing the traffic. I was wondering if we wanted to have any validation in the case that the user uses both methods.

Reply:
Discussed on a call: we decided that it's best to not get in the way of what the user configures, it's more likely that we'll cause issues than do anything meaningful to prevent or mitigate them